core/backend/src/workspace/Settings.ts

```diff
@@ -227,6 +218,7 @@ export class ITwinSettings implements Settings {
   }
 
   public addDictionary(dictionaryName: string, priority: SettingsPriority, settings: SettingDictionary) {
+    this.verifyPriority(priority);
```

Comment:
Could you explain to me why you raise the onSettingsChanged event in only one case but not the other? If we add the new dictionary through "this._dictionaries.splice(i, 0, file);" (because the new dictionary's priority was >= some preexisting dictionary's priority) the event doesn't get raised.

![image](https://user-images.githubusercontent.com/22119573/140062934-233e40e8-3657-4018-a992-e7283b72eca5.png)

Reply:
good catch. Fixed (and added test that shows it).